repair/repair_task.hh

```diff
@@ -252,6 +251,20 @@ protected:
     // TODO: implement abort and progress for data-sync repairs
 };
 
+class shard_repair_task_impl : public repair_task_impl {
+private:
+    lw_shared_ptr<repair_info> _ri;
+    std::exception_ptr _ex;
+public:
+    shard_repair_task_impl(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_id parent_id = task_id::create_null_id(), lw_shared_ptr<repair_info> ri = nullptr, std::exception_ptr ex = nullptr)
```

Comment:
Why so many default parameters? Is there any user relying on them? Is an object constructed with them valid?